### PR TITLE
Add origin to all outbound calls (AST-000)

### PR DESCRIPTION
**Pull request by greensd4**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/wrappers/client_test.go [hunk 1]
```diff
@@ -183,3 +184,26 @@ func TestGetAPIKeyPayload(t *testing.T) {
 		})
 	}
 }
+
+func TestSetAgentNameAndOrigin(t *testing.T) {
```
Comment (elchananarb) [inline internal/wrappers/client_test.go:188]: The TestSetAgentNameAndOrigin test only covers a positive scenario. It might be worth adding tests that verify proper handling when the values ​​are undefined or empty

File: internal/wrappers/client.go [hunk 2]
```diff
@@ -99,9 +99,12 @@ func retryHTTPRequest(requestFunc func() (*http.Response, error), retries int, b
 	return resp, nil
 }
 
-func setAgentName(req *http.Request) {
+func setAgentNameAndOrigin(req *http.Request) {
 	agentStr := viper.GetString(commonParams.AgentNameKey) + "/" + commonParams.Version
 	req.Header.Set("User-Agent", agentStr)
+
+	originStr := viper.GetString(commonParams.OriginKey)
+	req.Header.Set("origin", originStr)
```
Comment (elchananarb) [inline internal/wrappers/client.go:107]: Suggest changing to "Origin" (uppercase) to align with "User-Agent" and the test’s req.Header.Get("Origin")

**Review by elchananarb — APPROVED** [review]

File: internal/wrappers/client_test.go (hunk 1 above)
Comment (greensd4) [reply internal/wrappers/client_test.go:188]: there is always a default value for the origin

File: internal/wrappers/client.go (hunk 2 above)
Comment (greensd4) [reply internal/wrappers/client.go:107]: its not our choice. this is the header name expected by the server

File: internal/wrappers/client_test.go (hunk 1 above)
Comment (elchananarb) [reply internal/wrappers/client_test.go:188]: Maybe we should add a test to verify that if the Origin is undefined, we get 'CLI' as expected

**Review by elchananarb — APPROVED** [review]
